lib/helpers.js

```diff
@@ -66,38 +65,46 @@ export function setVerificationKeyPairId({verificationKeyPair, did}) {
     `${did}#${verificationKeyPair.publicKeyMultibase}`;
 }
 
-export function getKeyAgreementKeyPair({verificationKeyPair}) {
+export function getKeyAgreementKeyPair({verificationPublicKey}) {
   const contexts = [DID_CONTEXT_URL];
```

Comment:
This function is doing more than just getting a key agreement key pair and adding a context for just that key type ... it's also adding the DID context and the verification public key type context ... so here's what I recommend to fix that up:

We should pass `contexts` in here, I think ... and initialize it externally with `[DID_CONTEXT_URL, verificationPublicKey['@context']]`. Then in here we can push a context if we need to for the key agreement key. I think that would also allow us to remove the constants for `ED25519_KEY_2018_CONTEXT_URL` and the 2020 one.

We may need to export the `verificationPublicKey` with `context: true` option ... and then delete `@context` from the exported key, I'm not sure (to keep it inline with the previous behavior).

Reply:
Updated [8cff4bc](https://github.com/digitalbazaar/did-method-key/pull/55/commits/8cff4bcf4f7eb0f8361f1cd474551a8ab963c71d)